R/common.R

```diff
@@ -988,3 +988,77 @@ findBestAlternateScenarios <- function(dfAlternateFeat,
   dfOptResult
 }
 
+#' @title
+#' Generate ROC curve for a dataset.
+#' @description Generates ROC curve and AUC for Sensitivity/Specificity or 
+#' Precision/Recall.
+#' @param predictions A vector of predictions from a machine learning model.
+#' @param labels A vector of the true labels. Must be the same length as 
+#' predictions.
+#' @param aucType Can be "SS" or "PR". Defaults to SS.
+#' @param plotFlg Binary value controlling plots Defaults to FALSE (no).
+#' @return auc Integral AUC of ROC plot.
+#'
+#' @import ROCR
+#' @export
+#' @references \url{http://healthcare.ai}
+#' @seealso \code{\link{healthcareai}}
+#' @examples
+#' 
+#' # generate data
+#' df <- data.frame(a = rep( seq(0,1,by=0.1), times=5),
+#'                  b = rep(0,times=55))
+#' # add positive labels
+#' df[c(seq(11,55,by=11)),2] <- 1
+#' df[c(seq(10,55,by=11)),2] <- 1
+#' df[c(seq(9,55,by=22)),2] <- 1
+#' df[c(4, 18, 30, 39, 52 ),2 ] <- 1
+
+#' # prepare vectors
+#' pred <- df[,1]
+#' labels <- df[,2]
+#' 
+#' # generate the AUC
+#' auc = generateAUC(pred,labels,'SS','TRUE')
+#' 
+generateAUC <- function(predictions, labels, aucType, plotFlg) {
+  # Error check for uneven length predictions and labels
+  if (length(predictions) != length(labels)) {
+    stop('Data vectors are not equal length!')
+  }
+  
+  # use SS if aucType is missing
+  if (missing(aucType)) {
+    aucType <- 'SS'
+  }
+  
+  # plot ROC curve if flag is missing
+  if (missing(plotFlg)) {
+    plotFlg <- FALSE
+  }
+  
+  # generate ROC data
+  roc1 <- prediction( predictions, labels)
+  
+  # get performance and AUC from either curve type
+  if (aucType == 'PR') {
+    # ROC Curve Precision/Recall
+    perf <- performance(roc1, "prec", "rec")
+    perf.auc <- performance(roc1, measure = "auc")
+    cat("Area under the curve: ", perf.auc@y.values[[1]])
+  }
+  # defaults to SS
+  else {
```

Comment:
Since we can use a default aucType in the function definition, lets use an if else here. 

Perhaps if we always use if else then it'd save the 1 sec it takes to check--would love your thoughts on standards to set.